actionview/lib/action_view/helpers/url_helper.rb

```diff
@@ -327,13 +327,14 @@ def button_to(name = nil, options = nil, html_options = nil, &block)
         html_options = convert_options_to_data_attributes(options, html_options)
         html_options["type"] = "submit"
 
-        button = if block_given?
-          content_tag("button", html_options, &block)
+        content = if block_given?
+          capture(&block)
         else
-          html_options["value"] = name || url
-          tag("input", html_options)
+          name || url
         end
 
+        button = content_tag("button", content, html_options, &block)
```

Comment:
We've captured the block just above but now we're passing it along here too? Is it possible we can just let `content_tag` call `capture` and do this?

```suggestion
        button = content_tag("button", name || url, html_options, &block)
```